explorations/longmemeval/src/commands/prepare.ts

```diff
@@ -617,14 +619,43 @@ export class PrepareCommand {
     }
 
     // Apply session limit if specified
+    // IMPORTANT: Always include evidence sessions (answer_session_ids) to ensure the benchmark can succeed
     let sessionsToProcess = sessionsWithDates;
     if (options.sessionLimit) {
       const startIndex = processedSessionIds.size;
       const endIndex = Math.min(startIndex + options.sessionLimit, sessionsWithDates.length);
-      sessionsToProcess = sessionsWithDates.slice(0, endIndex);
-      console.log(
-        chalk.yellow(`\n📊 Processing limited to ${options.sessionLimit} sessions (${startIndex + 1} to ${endIndex})`),
+
+      // Get evidence session IDs that contain the answer
+      const evidenceSessionIds = new Set(question.answer_session_ids || []);
+
+      // Find which evidence sessions are NOT in the limited range
+      const sessionsInRange = sessionsWithDates.slice(0, endIndex);
+      const sessionIdsInRange = new Set(sessionsInRange.map(s => s.sessionId));
+
+      // Find evidence sessions that would be excluded
+      const excludedEvidenceSessions = sessionsWithDates.filter(
+        s => evidenceSessionIds.has(s.sessionId) && !sessionIdsInRange.has(s.sessionId),
       );
+
+      if (excludedEvidenceSessions.length > 0) {
+        // Include the excluded evidence sessions at the end
+        sessionsToProcess = [...sessionsInRange, ...excludedEvidenceSessions];
+        console.log(
+          chalk.yellow(
+            `\n📊 Processing ${sessionsToProcess.length} sessions (${options.sessionLimit} + ${excludedEvidenceSessions.length} evidence sessions)`,
+          ),
+        );
+        console.log(
+          chalk.gray(`   Evidence sessions included: ${excludedEvidenceSessions.map(s => s.sessionId).join(', ')}`),
+        );
+      } else {
+        sessionsToProcess = sessionsInRange;
+        console.log(
+          chalk.yellow(
+            `\n📊 Processing limited to ${options.sessionLimit} sessions (${startIndex + 1} to ${endIndex})`,
+          ),
+        );
+      }
```

Comment:
This stuff is here so that if you run a tiny benchmark like `--subset 1 --session-limit 3` it will make sure the evidence is always included in the history. Otherwise if you limit the sessions you might just not include the answers in the chat history so that the observer has a chance to make an observation on it. This is only here for small testing and doesn't do anything when running the whole dataset.